src/main/java/school/lemon/changerequest/java/generics/NumberGenerator/Printer.java

```diff
@@ -0,0 +1,33 @@
+package school.lemon.changerequest.java.generics.NumberGenerator;
+
+/**
+ * Created by lera on 27.12.16.
+ */
+public class Printer<T extends Number> {
+    NumberGenerator<T> numberGenerator;
+
+    public Printer(NumberGenerator numberGenerator) {
+        this.numberGenerator = numberGenerator;
+    }
+
+    public void generateAndPrint() throws ClassNotFoundException, IllegalAccessException, InstantiationException {
+        T[] array = numberGenerator.generateNumbers();
+        SumCalculator s1;
+        if (numberGenerator instanceof DoubleGenerator) {
+            s1 = new SumCalculatorDouble();
+        } else {
+            s1 = new SumCalculatorInteger();
+        }
+        System.out.println(s1.calculateSum(array));
+    }
+
+    public static void main(String[] args) throws IllegalAccessException, InstantiationException, ClassNotFoundException {
```

Comment:
Why do you need these exceptions `throws ClassNotFoundException, IllegalAccessException, InstantiationException`?